torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -563,799 +632,158 @@ TORCH_CUDA_CU_API c10::optional<ReductionParams> getNormalizationHeuristics(
   auto max_persistent_size =
       scheduler_utils::persistentBufferSize(fusion, runtime_info);
 
+  auto vectorizable_inputs_outputs =
+      scheduler_utils::getVectorizableInputsOutputs(first_red_tv);
+
+  // Vectorize as much as we can
+  size_t vectorize_factor = std::numeric_limits<size_t>::max();
+
+  for (auto tv : vectorizable_inputs_outputs) {
+    const auto tv_vectorize_factor = runtime_info.getVectorizableWidth(tv);
+    vectorize_factor = std::min(vectorize_factor, tv_vectorize_factor);
+  }
+
+  if (vectorize_factor == std::numeric_limits<size_t>::max()) {
+    vectorize_factor = 1;
+  }
+
   return NormalizationHeuristic(
       properties.reduction_numel,
       properties.iteration_numel,
       properties.fastest_dim_reduction,
       n_tensor_inputs,
       max_dtype_size,
       requires_persistence,
-      max_persistent_size);
+      max_persistent_size,
+      vectorize_factor);
 }
 
 TORCH_CUDA_CU_API c10::optional<ReductionParams> getNormalizationHeuristics(
     Fusion* fusion,
-    const at::ArrayRef<c10::IValue>& fusion_inputs) {
+    const at::ArrayRef<c10::IValue>& runtime_inputs) {
   FUSER_PERF_SCOPE("getNormalizationHeuristics");
-
-  SchedulerRuntimeInfo runtime_info(fusion, fusion_inputs, true);
-
+  SchedulerRuntimeInfo runtime_info(fusion, runtime_inputs, true);
   return getNormalizationHeuristics(fusion, runtime_info);
 }
+
 namespace {
 
 void schedulePersistentNormalization(
```

Comment:
Majority of logic has been moved to scheduler utils, it's now shared across reduction, multi-reduction, and normalization.